### PR TITLE
Change the default database synchronous configuration

### PR DESCRIPTION
**Pull request by nmkoremblum**
|Related issue|
|---|
|#22753|

## Description

Wazuh database synchronous mode has been changed to "1" (NORMAL) programmatically, as the configuration does not persist through sessions.

- Compilation without warnings in every supported platform
  - [X] Linux
- [x] Source installation
- [x] Package installation
- [x] Source upgrade
- [x] Package upgrade
- [x] Review logs syntax and correct language
- [x] QA templates contemplate the added capabilities: #22865

### CODE REVIEW

File: src/wazuh_db/wdb.c [hunk 1]
```diff
@@ -214,6 +214,7 @@ static const char *SQL_STMT[] = {
     [WDB_STMT_TASK_CANCEL_PENDING_UPGRADE_TASKS] = "UPDATE TASKS SET STATUS = '" WM_TASK_STATUS_CANCELLED "', LAST_UPDATE_TIME = ? WHERE NODE = ? AND STATUS = '" WM_TASK_STATUS_PENDING "' AND (COMMAND = 'upgrade' OR COMMAND = 'upgrade_custom');",
     [WDB_STMT_PRAGMA_JOURNAL_WAL] = "PRAGMA journal_mode=WAL;",
     [WDB_STMT_PRAGMA_ENABLE_FOREIGN_KEYS] = "PRAGMA foreign_keys=ON;",
+    [WDB_STMT_PRAGMA_SYNCHRONOUS_1] = "PRAGMA synchronous = 1;",
```
Comment (cborla) [inline src/wazuh_db/wdb.c]: We should keep the format similar to the other configurations, eliminating the spaces between the = and the value.
`[WDB_STMT_PRAGMA_SYNCHRONOUS_1] = "PRAGMA synchronous=1;",`

**Review by cborla — APPROVED** [review]
LGTM

**Comment by cborla** [comment]
## Branches 4.8.0 vs 22753-change-db-synchronous-mode

### Branch  22753-change-db-synchronous-mode

``` console
wdb-query2.py global "sql PRAGMA synchronous"
[
    {
        "synchronous": 1
    }
]


cat /var/ossec/etc/client.keys | wc -l
50000

TOKEN=$(curl -u wazuh:wazuh -k -X POST "https://localhost:55000/security/user/authenticate?raw=true")

curl -k -X POST "https://localhost:55000/groups" -H  "Authorization: Bearer $TOKEN" -H  "Content-Type: application/json" -d '{"group_id": "test_group10"}'

time curl -k -X PUT "https://localhost:55000/agents/group?wait_for_complete=true&group_id=test_group10" -H  "Authorization: Bearer $TOKEN"
{"data": {"affected_items": ["001",.... "50000", "50001"], "total_affected_items": 50001, "total_failed_items": 0, "failed_items": []}, "message": "All selected agents were assigned to test_group10", "error": 0}
real  7m49,405s
user  0m0,037s
sys 0m0,027s
```

### Branch 4.8.0

``` console
wdb-query2.py global "sql PRAGMA synchronous"
[
    {
        "synchronous": 2
    }
]


cat /var/ossec/etc/client.keys | wc -l
50000

TOKEN=$(curl -u wazuh:wazuh -k -X POST "https://localhost:55000/security/user/authenticate?raw=true")

curl -k -X POST "https://localhost:55000/groups" -H  "Authorization: Bearer $TOKEN" -H  "Content-Type: application/json" -d '{"group_id": "test_group10"}'

time curl -k -X PUT "https://localhost:55000/agents/group?wait_for_complete=true&group_id=test_group10" -H  "Authorization: Bearer $TOKEN"
{"data": {"affected_items": ["001",.... "50000", "50001"], "total_affected_items": 50001, "total_failed_items": 0, "failed_items": []}, "message": "All selected agents were assigned to test_group10", "error": 0}
real  11m23,399s
user  0m0,028s
sys 0m0,049s
```

### Conclusion

| Branch | synchronous value | Time it takes to perform group assignment|
|-|-|-|
|[22753-change-db-synchronous-mode](https://github.com/wazuh/wazuh/tree/22753-change-db-synchronous-mode)| NORMAL (1) | 7m 49s |
|[4.8.0](https://github.com/wazuh/wazuh/tree/4.8.0)| FULL (2) | 11m 23s |

In this case the difference between versions was about 30% less time that `branch 22753-change-db-synchronous-mode` took than branch `4.8.0`.

**Review by vikman90 — CHANGES_REQUESTED** [review]

**Comment by cborla** [comment]
## Time a version takes to perform group assignment (Vulnerability detector disabled)

### Branch  22753-change-db-synchronous-mode

Disabling vulnerability detection from ossec.conf

``` xml
<vulnerability-detection>
  <enabled>no</enabled>
```

#### Test:
``` console
wdb-query2.py global "sql PRAGMA synchronous"
[
    {
        "synchronous": 1
    }
]

cat /var/ossec/etc/client.keys | wc -l
50000

TOKEN=$(curl -u wazuh:wazuh -k -X POST "https://localhost:55000/security/user/authenticate?raw=true")

curl -k -X POST "https://localhost:55000/groups" -H  "Authorization: Bearer $TOKEN" -H  "Content-Type: application/json" -d '{"group_id": "test_group3"}'

time curl -k -X PUT "https://localhost:55000/agents/group?wait_for_complete=true&group_id=test_group3" -H  "Authorization: Bearer $TOKEN"
{"data": {"affected_items": ["001",.... "50000", "50001"], "total_affected_items": 50001, "total_failed_items": 0, "failed_items": []}, "message": "All selected agents were assigned to test_group3", "error": 0}
real  5m13,638s
user  0m0,020s
sys 0m0,021s
```

### Branch v4.8.0-beta4

Disabling vulnerability detection from ossec.conf
``` xml
<vulnerability-detection>
  <enabled>no</enabled>
```

#### Test:
``` console
/var/ossec/bin/wazuh-control info
WAZUH_VERSION="v4.8.0"
WAZUH_REVISION="40806"
WAZUH_TYPE="server"

wdb-query2.py global "sql PRAGMA synchronous"
[
    {
        "synchronous": 2
    }
]

cat /var/ossec/etc/client.keys | wc -l
50000

TOKEN=$(curl -u wazuh:wazuh -k -X POST "https://localhost:55000/security/user/authenticate?raw=true")

curl -k -X POST "https://localhost:55000/groups" -H  "Authorization: Bearer $TOKEN" -H  "Content-Type: application/json" -d '{"group_id": "test_group3"}'

time curl -k -X PUT "https://localhost:55000/agents/group?wait_for_complete=true&group_id=test_group3" -H  "Authorization: Bearer $TOKEN"
{"data": {"affected_items": ["001",.... "50000", "50001"], "total_affected_items": 50001, "total_failed_items": 0, "failed_items": []}, "message": "All selected agents were assigned to test_group3", "error": 0}
real  6m50,241s
user  0m0,020s
sys 0m0,025s
```

### Branch v4.8.0-beta5

Disabling vulnerability detection from ossec.conf

``` xml
<vulnerability-detection>
  <enabled>no</enabled>
```
#### Test:
``` console
/var/ossec/bin/wazuh-control info
WAZUH_VERSION="v4.8.0"
WAZUH_REVISION="40807"
WAZUH_TYPE="server"

wdb-query2.py global "sql PRAGMA synchronous"
[
    {
        "synchronous": 2
    }
]

cat /var/ossec/etc/client.keys | wc -l
50000

TOKEN=$(curl -u wazuh:wazuh -k -X POST "https://localhost:55000/security/user/authenticate?raw=true")

curl -k -X POST "https://localhost:55000/groups" -H  "Authorization: Bearer $TOKEN" -H  "Content-Type: application/json" -d '{"group_id": "test_group3"}'

time curl -k -X PUT "https://localhost:55000/agents/group?wait_for_complete=true&group_id=test_group3" -H  "Authorization: Bearer $TOKEN"
{"data": {"affected_items": ["001",.... "50000", "50001"], "total_affected_items": 50001, "total_failed_items": 0, "failed_items": []}, "message": "All selected agents were assigned to test_group3", "error": 0}
real  6m38,176s
user  0m0,027s
sys 0m0,022s
```

### Conclusion

| Branch | synchronous value | Time it takes to perform group assignment|
|-|-|-|
|[22753-change-db-synchronous-mode](https://github.com/wazuh/wazuh/tree/22753-change-db-synchronous-mode)| NORMAL (1) | 5m 13s |
|v4.8.0-beta4| FULL (2) | 6m 50s |
|v4.8.0-beta5| FULL (2) | 6m 38s |

In this case the difference between versions was about 20% less time that `branch 22753-change-db-synchronous-mode` took than branch `v4.8.0-betaX`.

**Review by vikman90 — COMMENTED** [review]
## Tests

### 4.8.0 (unfixed)

|Query|Result|
|--|--|
|`global sql PRAGMA synchronous`|2|

### This branch (fixed)

|Query|Result|
|--|--|
|`global sql PRAGMA synchronous`|1|
|`agent 000 sql PRAGMA synchronous`|2|

**Pull request merged**